src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1021,6 +1022,16 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
                             Debug.Assert(connection is not null || !_http2Enabled);
                             if (connection is not null)
                             {
+                                if (request.IsWebSocketH2Request())
+                                {
+                                    if (await connection.InitialSettingsReceived.WaitWithCancellationAsync(cancellationToken).ConfigureAwait(false) && !connection.IsConnectEnabled)
+                                    {
+                                        HttpRequestException exception = new("Extended CONNECT is not supported");
+                                        exception.Data["SETTINGS_ENABLE_CONNECT_PROTOCOL"] = false;
```

Comment:
What is the benefit of this if we're always storing `false`?

Reply:
We are adding key "SETTINGS_ENABLE_CONNECT_PROTOCOL" into the dictionary, that can be later checked to differ disabled extended connect vs other errors. Do you mean that it can be more compact?